apps/www/_blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14.mdx

```diff
@@ -0,0 +1,192 @@
+  alt="Tweet from Lee Robinson about Next.js 14 containing no new APIs - dark mode version"
+  src="/images/blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14/lee-tweet-dark.png"
+/>
+
+That's right, the headline feature is no new features!
+
+This may sound underwhelming at first, but is incredibly good news for the stability of Next.js. This release comes with a huge number of performance and stability improvements - such as [Server Actions](https://nextjs.org/docs/app/api-reference/functions/server-actions) being marked as stable. This means we can finally start promoting this fantastically simple way of authenticating users—entirely server-side!
+
+```tsx
+export default async function Page() {
+  const signIn = async () => {
+    'use server'
+    supabase.auth.signInWithOAuth({...})
+  }
+
+  return (
+    <form action={signIn}>
+      <button>Sign in with GitHub</button>
+    </form>
+  )
+}
+```
+
+With [Server Components](https://nextjs.org/docs/app/building-your-application/rendering/server-components), fetching data in Next.js became as simple as:
+
+```tsx
+export default async function Page() {
+  const { data } = await supabase.from('...').select()
+  return ...
+}
+```
+
+Now with Server Actions, mutation logic can be co-located with the Server Components that fetch data and render the page:
+
+```tsx
+export default async function Page() {
+  const { data } = await supabase.from('...').select()
+
+  const createNote = async () => {
+    'use server'
+    await supabase.from('...').insert({...})
+  }
+
+  return ...
+}
+```
+
+> To hear more about our thoughts on Next.js Conf and the release of Next.js 14, check out our [Twitter space](https://twitter.com/i/spaces/1yoKMwNWbRjJQ?s=20). [Yuri](https://twitter.com/yuricodesbot), [Alaister](https://twitter.com/alaisteryoung), [Terry](https://twitter.com/saltcod) and [myself](https://twitter.com/jonmeyers_io) talk through how we use Next.js at Supabase and what we personally found most exciting about Next.js Conf and the release of Next.js 14.
+
+## Is Supabase compatible with Next.js 14?
+
+Yes, it is! So much so that [Guillermo Rauch](https://twitter.com/rauchg) shouted us out in the keynote!
+
+<img
+  className="dark:hidden !m-0"
+  alt="Tweet from Guillermo Rauch showing Supabase as one of the companies compatible with the Next.js App Router - light mode version"
+  src="/images/blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14/guillermo-tweet-light.png"
+/>
+
+<img
+  className="hidden dark:block !m-0"
+  alt="Tweet from Guillermo Rauch showing Supabase as one of the companies compatible with the Next.js App Router - dark mode version"
+  src="/images/blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14/guillermo-tweet-dark.png"
+/>
+
+Since the release of the [App Router](https://nextjs.org/docs/app) - launched as beta with Next.js 13 - we have been working closely with the team at Vercel to ensure that Supabase is fully compatible with every part of Next.js.
```

Comment:
```suggestion
Since the release of the [App Router](https://nextjs.org/docs/app)—launched as beta with Next.js 13—we have been working closely with the team at Vercel to ensure that Supabase is fully compatible with every part of Next.js.
```